tests/generation/test_generation_tf_logits_process.py

```diff
@@ -331,6 +334,86 @@ def test_forced_eos_token_logits_processor(self, use_xla):
         scores = logits_processor(input_ids, scores, cur_len)
         self.assertFalse(tf.math.reduce_any(tf.math.is_inf((scores))))
 
+    @parameterized.expand([(False,), (True,)])
+    def test_suppress_tokens_at_begin_logits_processor(self, use_xla):
+        vocab_size = 20
+        batch_size = 4
+
+        begin_suppress_tokens = [1, 2, 3]
+        begin_index = 5
+
+        logits_processor = TFSuppressTokensAtBeginLogitsProcessor(
+            begin_suppress_tokens=begin_suppress_tokens, begin_index=begin_index
+        )
+        if use_xla:
+            logits_processor = tf.function(logits_processor, jit_compile=True)
+
+        # Check that no scores are suppressed if begin_index is not reached
+        cur_len = 4
+        input_ids = ids_tensor((batch_size, cur_len), vocab_size=20)
+        scores = self._get_uniform_logits(batch_size, vocab_size)
+        scores = logits_processor(input_ids, scores, cur_len)
+        self.assertFalse(tf.math.reduce_any(tf.math.is_inf((scores))))
+
+        # Check that scores are suppressed if begin_index is reached
+        cur_len = 5
+        input_ids = ids_tensor((batch_size, cur_len), vocab_size=20)
+        scores = self._get_uniform_logits(batch_size, vocab_size)
+        scores = logits_processor(input_ids, scores, cur_len)
+        self.assertTrue(tf.math.reduce_all(tf.math.is_inf(tf.gather(scores, begin_suppress_tokens, axis=1))))
+
+    @parameterized.expand([(False,), (True,)])
+    def test_suppress_tokens_logits_processor(self, use_xla):
+        vocab_size = 20
+        batch_size = 4
+
+        suppress_tokens = [1, 3, 5]
+        keep_tokens = [i for i in range(vocab_size) if i not in suppress_tokens]
+
+        logits_processor = TFSuppressTokensLogitsProcessor(suppress_tokens=suppress_tokens)
+        if use_xla:
+            logits_processor = tf.function(logits_processor, jit_compile=True)
+
+        # Check that suppress_tokens are suppressed and others are not
+        cur_len = 5
+        input_ids = ids_tensor((batch_size, cur_len), vocab_size=20)
+        scores = self._get_uniform_logits(batch_size, vocab_size)
+        scores = logits_processor(input_ids, scores, cur_len)
+        self.assertTrue(tf.math.reduce_all(tf.math.is_inf(tf.gather(scores, suppress_tokens, axis=1))))
+        self.assertFalse(tf.math.reduce_any(tf.math.is_inf(tf.gather(scores, keep_tokens, axis=1))))
```

Comment:
Very nice tests, the pytorch version doesn't have these! I should open a PR to add that! 👍🏻